### ISSUE TITLE
ModuleConcatenationPlugin: Unexpected token

### ISSUE THREAD

**Issue by screendriver** [webpack/webpack#5304]
**Do you want to request a *feature* or report a *bug*?** bug

**What is the current behavior?**

When I include the `ModuleConcatenationPlugin` I get following error:

```sh
ERROR in chunk main [initial]
[name].[chunkhash].js
Unexpected token (61:45)
```

Without the plugin everything works fine.

**If the current behavior is a bug, please provide the steps to reproduce.**

I created a [Gist](https://gist.github.com/screendriver/b4feadaf456ad0a3a9be38015e638cdd) that shows my `webpack.config.js`

**What is the expected behavior?**

It should build without errors.

**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**

Node 8.1.4
webpack 3.3.0
macOS sierra

**Comment by sokra** [comment]
I need to info to address this. What's in line 61 of the output file?

**Comment by screendriver** [comment]
My main bundle will not be generated so I can't tell you what's in line 61:

```sh
Hash: 98fccd7c07eedd1d2c06                                                              
Version: webpack 3.3.0
Time: 13979ms
                                    Asset       Size  Chunks                    Chunk Names
     34aa91b8b2a3677f3d168feed62deffb.svg    5.48 kB          [emitted]         
                0.f16762b43f6758949a36.js     260 kB       0  [emitted]  [big]  
                1.18fe110d1a6f16779c87.js    4.16 kB       1  [emitted]         
           vendor.b4bcbda5132dfa0e9d87.js     296 kB       2  [emitted]  [big]  vendor
         manifest.9d174e32da288a654e18.js    1.47 kB       4  [emitted]         manifest
561f831e12acf14208223d59c171bb60.main.css    1.94 kB       3  [emitted]         main
                               index.html  590 bytes          [emitted]         
   [8] ./src/shared/styles/colors.ts 2.25 kB {3} [built]
   [9] ./src/shared/styles/distances.ts 1.1 kB {3} [built]
  [17] (webpack)/buildin/global.js 509 bytes {2} [built]
  [19] ./node_modules/react-redux/es/index.js + 14 modules 37.6 kB {2} [built]
  [32] ./node_modules/redux/es/index.js + 6 modules 21.1 kB {2} [built]
  [34] ./src/shared/types/travelPeriod.ts 2.29 kB {3} [built]
  [44] (webpack)/buildin/module.js 517 bytes {2} [built]
  [45] ./src/shared/services/travelPeriodService.ts 5.82 kB {3} [built]
  [65] ./node_modules/lodash-es/isPlainObject.js + 8 modules 5.71 kB {2} [built]
  [67] ./src/shared/services/durationService.ts 1.18 kB {3} [built]
 [117] multi ./src/index 28 bytes {3} [built]
 [118] ./src/index.tsx + 53 modules 101 kB {3} [built]
 [230] ./src/assets/css/normalize.css 41 bytes {3} [built]
 [239] ./src/shared/ui/hc-logo/hc-logo-wb.svg?asPath 82 bytes {3} [built]
 [375] ./node_modules/css-loader!./src/assets/css/normalize.css 2.11 kB [built]
    + 361 hidden modules

ERROR in chunk main [initial]
[name].[chunkhash].js
Unexpected token (61:45)
Child html-webpack-plugin for "index.html":
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 665 bytes {0} [built]
       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
        + 1 hidden module
Child html-webpack-plugin for "index.html":
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.html 665 bytes {0} [built]
       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin /Users/me/development/project/node_modules/extract-text-webpack-plugin/dist /Users/me/development/project/node_modules/css-loader/index.js!/Users/me/development/project/src/assets/css/normalize.css:
       [0] ./node_modules/css-loader!./src/assets/css/normalize.css 2.11 kB {0} [built]
        + 1 hidden module
make: *** [build] Error 2
error Command failed with exit code 2.
```

**Comment by sokra** [comment]
Could you try running with `--display-error-details` resp. `stats: { errorDetails: true }`

**Comment by screendriver** [comment]
```sh
ERROR in chunk main [initial]
[name].[chunkhash].js
Unexpected token (61:45)
SyntaxError: Unexpected token (61:45)
    at Parser.pp$4.raise (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2610:13)
    at Parser.pp.unexpected (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:637:8)
    at Parser.pp$3.parseExprAtom (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2094:10)
    at Parser.pp$3.parseExprSubscripts (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1955:19)
    at Parser.pp$3.parseMaybeUnary (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1932:17)
    at Parser.pp$3.parseExprOps (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1874:19)
    at Parser.pp$3.parseMaybeConditional (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1857:19)
    at Parser.pp$3.parseMaybeAssign (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1832:19)
    at Parser.pp$3.parseExprList (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2528:20)
    at Parser.pp$3.parseExprAtom (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2073:26)
    at Parser.pp$3.parseExprSubscripts (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1955:19)
    at Parser.pp$3.parseMaybeUnary (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1932:17)
    at Parser.pp$3.parseExprOps (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1874:19)
    at Parser.pp$3.parseMaybeConditional (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1857:19)
    at Parser.pp$3.parseMaybeAssign (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1832:19)
    at Parser.pp$3.parseExpression (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1803:19)
    at Parser.pp$1.parseReturnStatement (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:916:31)
    at Parser.pp$1.parseStatement (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:764:35)
    at Parser.pp$1.parseSwitchStatement (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:952:34)
    at Parser.pp$1.parseStatement (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:765:35)
    at Parser.pp$1.parseBlock (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1076:23)
    at Parser.pp$3.parseFunctionBody (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2466:22)
    at Parser.pp$1.parseFunction (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1177:8)
    at Parser.pp$3.parseExprAtom (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2082:17)
    at Parser.pp$3.parseExprSubscripts (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1955:19)
    at Parser.pp$3.parseMaybeUnary (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1932:17)
    at Parser.pp$3.parseExprOps (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1874:19)
    at Parser.pp$3.parseMaybeConditional (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1857:19)
    at Parser.pp$3.parseMaybeAssign (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:1832:19)
    at Parser.pp$3.parseExprList (/Users/me/development/project/node_modules/webpack/node_modules/acorn/dist/acorn.js:2528:20)
```

**Comment by sokra** [comment]
hmm... Could you add `Error.stackTraceLimit = 1000;` to your webpack.config.js, to get a longer stack trace.

**Comment by screendriver** [comment]
Unfortunately that didn't change anything.

I set `Error.stackTraceLimit = 1000;` at the top of my file like

```js
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer')
  .BundleAnalyzerPlugin;
const CleanWebpackPlugin = require('clean-webpack-plugin');
const DashboardPlugin = require('webpack-dashboard/plugin');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const merge = require('webpack-merge');
const path = require('path');
const webpack = require('webpack');
const ExtractTextPlugin = require('extract-text-webpack-plugin');

Error.stackTraceLimit = 1000;
const isDev = process.env.NODE_ENV !== 'production';

const common = {
  // ...
}
```

Is this correctly added by my side?

**Comment by ooflorent** [comment]
Fixed in #5338
